src/Phinx/Db/Plan/Plan.php

```diff
@@ -206,6 +216,90 @@ protected function forgetTable(Table $table, $actions)
         return $result;
     }
 
+    /**
+     * Finds all DropForeignKey actions in an AlterTable and moves
+     * all conflicting DropIndex action in `$this->indexes` into the
+     * given AlterTable.
+     *
+     * @param AlterTable $alter The collection of actions to inspect
+     * @return AlterTable The updated AlterTable object. This function
+     * has the side effect of changing the `$this->indexes` property.
+     */
+    protected function remapContraintAndIndexConflicts(AlterTable $alter)
+    {
+        $newAlter = new AlterTable($alter->getTable());
+        collection($alter->getActions())
+            ->unfold(function ($action) {
+                if (!$action instanceof DropForeignKey) {
+                    return [$action];
+                }
+
+                list($this->indexes, $dropIndexActions) = $this->forgetDropIndex(
+                    $action->getTable(),
+                    $action->getForeignKey()->getColumns(),
+                    $this->indexes
+                );
+
+                if (!empty($dropIndexActions)) {
+                    return array_merge([$action], $dropIndexActions);
+                }
+
+                return [$action];
+            })
+            ->each(function ($action) use ($newAlter) {
+                $newAlter->addAction($action);
+            });
+
+        return $newAlter;
+    }
+
+    /**
+     * Deletes any DropIndex actions for the given table and exact columns
+     *
+     * @param Table $table The table to find in the list of actions
```

Comment:
Your IDE could be configured to auto fqcn here in doc blocks. Not sure if our sniffer for 3.x can fix this yet on his own.

Reply:
I thin in phinx in general we are not using fqcn

Reply:
Probably just a matter of proper coding standards being applied via auto-fixer.
Can also be done laters.